tests/dll-test.c

```diff
@@ -13,10 +13,14 @@
 __declspec(dllimport) int __cdecl add_c_export(int a, int b);
 __declspec(dllimport) int __stdcall add_std_export(int a, int b);
 
+#if !defined(__CYGWIN__)
```

Comment:
I would create two separate functions for CYGWIN and MINGW

Reply:
Can you be more specific what do you mean by "functions", please?

Reply:
separate test functions test_func_pointer_mingw and test_func_pointer_cygwin without ifdef inside.